llm/api.py

```diff
@@ -38,35 +48,112 @@ def create_chat(request):
+                }
+            ],
+            functions=[
+                {
+                    "name": "detect_languages",
+                    "description": "Detecting language and other insights on a piece of user input text.",
+                    "parameters": {
+                        "type": "object",
+                        "properties": {
+                            "language": {
+                                "title": "Language",
+                                "description": "The primary detected language e.g English, French, Hindi, etc",
+                                "type": "string",
+                            },
+                            "confidence": {
+                                "title": "Confidence",
+                                "description": "Confidence level of the language detection from scale of 0 to 1",
+                                "type": "number",
+                            },
+                            "english_translation": {
+                                "title": "English translation",
+                                "description": "English translation of the user input text if not in English",
+                                "type": "string",
+                            },
+                            "translation_confidence": {
+                                "title": "Confidence",
+                                "description": "Confidence level of the language translation to English from scale of 0 to 1",
+                                "type": "number",
+                            },
+                        },
+                        "required": [
+                            "language",
+                            "confidence",
+                            "english_translation",
+                            "translation_confidence",
+                        ],
+                    },
+                }
+            ],
+            function_call={"name": "detect_languages"},
+            temperature=0,
+        )
 
-        response = run_chat_chain(
-            prompt=prompt,
-            session_id=session_id,
-            primary_language=primary_language,
-            english_translation_prompt=english_translation_prompt,
-            organization_id=organization.id,
-            gpt_model=gpt_model,
+        language_results = json.loads(
+            response["choices"][0]["message"]["function_call"]["arguments"]
+        )
+        logger.info("Fetched language results via function calls")
+        logger.info("language detected", language_results["language"])
+
+        # 2. Pull relevant chunks from vector database
+        prompt_embeddings = openai.Embedding.create(
+            model="text-embedding-ada-002", input=prompt
+        )["data"][0]["embedding"]
+
+        results = Embedding.objects.alias(
+            distance=L2Distance("text_vectors", prompt_embeddings)
+        ).filter(distance__gt=0.7)
```

Comment:
What are your thoughts abou a later feature allowing users to pass this threshold via the chat request body? We experimentally and kinda arbitrarily came to `0.7` but I imagine different tenants would like to tune this.

Reply:
We can always make this parameter tunable but do the NGOs understand interpretation of that number ? If yes, we can parameterize this as our input. 
Otherwise we wait till we start hearing that this is becoming a problem and not giving correct results.

Reply:
Agreed, makes sense